src/mds/SimpleLock.h

```diff
@@ -359,13 +359,23 @@ class SimpleLock {
     return get_sm()->states[state].can_wrlock == ANY ||
       (get_sm()->states[state].can_wrlock == AUTH && parent->is_auth()) ||
       (get_sm()->states[state].can_wrlock == XCL && client >= 0 && (get_xlock_by_client() == client ||
-								    get_excl_client() == client));
+                                                                    (get_excl_client() == client &&
+                                                                     !(parent->is_dir() &&
+                                                                       (type->type == CEPH_LOCK_IFILE ||
+                                                                        type->type == CEPH_LOCK_IAUTH ||
+                                                                        type->type == CEPH_LOCK_ILINK ||
+                                                                        type->type == CEPH_LOCK_IXATTR)))));
```

Comment:
Is this problem actually unique to directories? Don't we want to revoke `Fx` even for regular files?

Reply:
Only for parent directory will have this issue.